Code/GraphMol/FileParsers/XYZFileParser.cpp

```diff
@@ -0,0 +1,177 @@
+        throw FileParseException(errout.str());
+    }
+    size_t delim5 = atomLine.find_first_of(whitespace, delim4);
+    if (delim5 == std::string_view::npos) {
+        std::ostringstream errout;
+        errout << "Missing coordinates on line " << line << std::endl;
+        throw FileParseException(errout.str());
+    }
+    size_t delim6 = atomLine.find_first_not_of(whitespace, delim5);
+    if (delim6 == std::string_view::npos) {
+        std::ostringstream errout;
+        errout << "Missing coordinates on line " << line << std::endl;
+        throw FileParseException(errout.str());
+    }
+    size_t delim7 = atomLine.find_last_not_of(whitespace) + 1;
+    
+    // set conformer
+    try {
+        pos.x = FileParserUtils::toDouble(atomLine.substr(delim2, delim3 - delim2), false);
+    } catch (boost::bad_lexical_cast &) {
+        std::ostringstream errout;
+        errout << "Cannot convert '" << atomLine.substr(delim2, delim3 - delim2) << "' to double on line " << line << std::endl;
+        throw FileParseException(errout.str());
+    }
+
+    try {
+        pos.y = FileParserUtils::toDouble(atomLine.substr(delim4, delim5 - delim4), false);
+    } catch (boost::bad_lexical_cast &) {
+        std::ostringstream errout;
+        errout << "Cannot convert '" << atomLine.substr(delim4, delim5 - delim4) << "' to double on line " << line << std::endl;
+        throw FileParseException(errout.str());
+    }
+    
+    try {
+        pos.z = FileParserUtils::toDouble(atomLine.substr(delim6, delim7 - delim6), false);
+    } catch (boost::bad_lexical_cast &) {
+        std::ostringstream errout;
+        errout << "Cannot convert '" << atomLine.substr(delim6, delim7 - delim6) << "' to double on line " << line << std::endl;
+        throw FileParseException(errout.str());
+    }
+    
+    // create atom
+    Atom *atom = new Atom;
+    
+    std::string symb{atomLine.substr(delim0, delim1 - delim0)};
+    if (symb.size() == 2 && symb[1] >= 'A' && symb[1] <= 'Z') {
+        symb[1] = static_cast<char>(tolower(symb[1]));
+    }
+    
+    try {
+      atom->setAtomicNum(PeriodicTable::getTable()->getAtomicNumber(symb));
+    } catch (const Invar::Invariant &e) {
+      delete atom;
+      throw FileParseException(e.what());
+    }
+    
+    return atom;
+}
+
+
+RWMol *XYZDataStreamToMol(std::istream &inStream) {
+    PRECONDITION(inStream, "no stream");
+    unsigned int numAtoms = 0;
+    
+    std::string_view num{getLine(inStream)};
+    try {
+        numAtoms = FileParserUtils::toUnsigned(num);
+    } catch (boost::bad_lexical_cast &) {
+        std::ostringstream errout;
+        errout << "Unable to recognize the number of atoms: cannot convert '" << num << "' to unsigned int on line 0" << std::endl;
+        throw FileParseException(errout.str());
+    }
+    
+    std::string comment{getLine(inStream)};
+    
+    RWMol *mol = nullptr;
+    if (numAtoms != 0) {
+        mol = new RWMol();
```

Comment:
As written, this will return a nullptr if there are zero atoms in the input. 

File formats like Mol/SDF allow zero-atom molecules and the RDKit supports that, so you might think about doing something like:
```suggestion
    RWMol *mol = new RWMol();
    if  (numAtoms) {
```


BTW: adding a test for this edge case would be a good idea.